cleanbuild: set the container language to C.UTF-8

elopio opened the pull request:
LP: [#1662456](https://bugs.launchpad.net/snapcraft/+bug/1662456)

ogra1 commented:
shouldn't this be C.UTF-8 (which is always available, contrary to en_*.UTF-8 which needs to be generated (admittedly often by default, but there in no guarantee))

sergiusens commented:
On Wed, May 10, 2017 at 7:10 AM, Oliver Grawert <notifications@github.com>
wrote:

> shouldn't this be C.UTF-8 (which is always available, contrary to
> en_*.UTF-8 which needs to be generated (admittedly often by default, but
> there in no guarantee))
>
yes it should. Our travis thing should also move to this (diff PR) to not
localegen all the time.

elopio commented:
That works. Thanks for the reviews.

elopio commented:
@kalikiana copying here what I mentioned during the stand up. Two problems to add a test here:

Currently, the integration tests for cleanbuild is disabled. We need to reenable it first.
But even with that test working, it takes a long time to start the container. I think it's not worth it to add a slow test to check this condition. I tested it manually, added a unit test to pin the fact that we are setting the environment, and in #1309 (thanks to your suggestion) I added an integration test for a non-ascii desktop file using the C locale. I think that's good enough.

sergiusens commented:
@kalikiana mind readressing your review after the comments?

@elopio and @kalikiana a test in manual-tests.md would be enough IMO for this particular case so it is at least documented and we have a path to automation.

elopio commented on manual-tests.md:
```diff
@@ -42,6 +42,12 @@
+# Test cleanbuild with non-ascii characters in the desktop file.
+
+1. Run `snapcraft cleanbuild` for the snap in `integration_tests/snaps/desktop-with-non-ascii`
+   * Check that the build succeeds and you get the `.snap` file.
```
This snap is added by #1309

elopio commented:
@kalikiana please re-review this.
